online-shop/src/app/modules/shopping-cart/components/presentational/shopping-cart-details-view/shopping-cart-details-view.component.ts

```diff
@@ -0,0 +1,15 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { Product } from 'src/app/modules/shared/types/products.types';
+
+@Component({
+  selector: 'app-shopping-cart-details-view',
+  templateUrl: './shopping-cart-details-view.component.html',
+  styleUrls: ['./shopping-cart-details-view.component.scss']
+})
+export class ShoppingCartDetailsViewComponent implements OnInit{
+    @Input() cartProducts!: Product[];
+   
+    ngOnInit(): void {
+      this.cartProducts.forEach(product=>(product.quantity=0));
```

Comment:
Seems like you are doing some logic in a presentational view. This should be done in /containers/ and "modify" your mock here, and pass them "modified" to the presentational layer